scripts/cmake/common.cmake

```diff
@@ -72,10 +72,14 @@ endmacro()
 
 if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
     add_compiler_flags(-Werror)
-    add_compiler_flags(-pedantic)
-    add_compiler_flags(-pedantic-errors)
-    add_compiler_flags(-fvisibility=hidden)
     add_compiler_flags(-fstrict-aliasing)
+
+    # The following options are not valid when clang-cl is used.
+    if(UNIX)
```

Comment:
I guess this check could be ```UNIX OR CMAKE_CXX_COMPILER_ID MATCHES "GNU"``` since we use MinGW on windows

Reply:
I did `NOT MSVC` which should handle all cases. It might be too extensive but we can solve that once we actually encounter issues with this approach.